wrapper/minecraft/src/main/kotlin/org/quiltmc/qkl/wrapper/minecraft/text/TextDsl.kt

```diff
@@ -0,0 +1,210 @@
+     *
+     * @author NoComment1105
+     */
+    public inline fun nbt(
+        pathPattern: String,
+        interpreting: Boolean,
+        separator: Optional<Text>,
+        nbt: TextData,
+        action: MutableStyle.() -> Unit = { }
+    ) {
+        val mutableStyle = MutableStyle()
+        mutableStyle.apply(action)
+        text.append(mutableStyle.applyTo(Text.nbt(pathPattern, interpreting, separator, nbt)))
+    }
+
+    /**
+     * @see Text.nbt
+     *
+     * @author NoComment1105
+     */
+    public inline fun nbt(
+        pathPattern: String,
+        interpreting: Boolean,
+        separator: Optional<Text>,
+        noinline nbt: ((ServerCommandSource) -> Stream<NbtCompound>),
+        action: MutableStyle.() -> Unit = { }
+    ) {
+        val mutableStyle = MutableStyle()
+        mutableStyle.apply(action)
+        text.append(mutableStyle.applyTo(Text.nbt(pathPattern, interpreting, separator, nbt)))
+    }
+
+    /**
+     * Add a plain text
+     *
+     * @param value The text to add
+     * @see MutableStyle for action
+     *
+     * @author NoComment1105
+     */
+    public inline fun text(value: String, action: MutableStyle.() -> Unit = { }) {
+        val mutableStyle = MutableStyle()
+        mutableStyle.apply(action)
+        text.append(mutableStyle.applyTo(Text.empty().append(value)))
+    }
+
+    /**
+     * Adds a scoreboard
+     *
+     * @param name The name to add to the scoreboard
+     * @param objective The objective of the scoreboard
+     * @see MutableStyle for action
+     *
+     * @author NoComment1105
+     */
+    public inline fun scoreboardText(
+        name: String,
+        objective: String,
+        action: MutableStyle.() -> Unit = { }
+    ) {
+        val mutableStyle = MutableStyle()
+        mutableStyle.apply(action)
+        text.append(mutableStyle.applyTo(Text.score(name, objective)))
+    }
+
+    /**
+     * Adds a resolvable entity selector
+     *
+     * @param separator the optional separator if there's multiple matches issued from the selector
+     * @param selector the selector
+     * @see MutableStyle for action
+     *
+     * @author NoComment1105
+     */
+    public inline fun selectorText(
+        selector: String,
+        separator: Optional<Text>,
+        action: MutableStyle.() -> Unit = { }
+    ) {
+        val mutableStyle = MutableStyle()
+        mutableStyle.apply(action)
+        this.text.append(mutableStyle.applyTo(Text.selector(selector, separator)))
+    }
+
+    /**
+     * Adds a mutable empty text
+     *
+     * @see MutableStyle for action
+     *
+     * @author NoComment1105
+     */
+    public inline fun empty(action: MutableStyle.() -> Unit = { }) {
+        val mutableStyle = MutableStyle()
+        mutableStyle.apply(action)
+        text.append(mutableStyle.applyTo(Text.empty()))
+    }
+
+    /**
+     * Gets the text from the DSL
+     *
+     * @author NoComment1105
+     */
+    public fun getText(): Text {
```

Comment:
i dunno what's wrong with this but it just looks wrong for me. it really feels like it should be a (perhaps internal) property